js/client-chat-tournament.js

```diff
@@ -208,13 +208,7 @@
 				return;
 			}
 
-			var teamIndex = undefined;
-			if (!forceFormatChange && this.$teamSelect.children().val()) {
-				teamIndex = parseInt(this.$teamSelect.children().val());
-				if (isNaN(teamIndex)) teamIndex = undefined;
-			}
-
-			this.$teamSelect.html(app.rooms[''].renderTeams(this.info.format, teamIndex));
+			this.$teamSelect.html(app.rooms[''].renderTeams.call(this, this.info.format));
```

Comment:
I do find calling `updateTeams` on every join and leave superfluous but having some time to select a team before the tournament starts seems better than not.

How about we make it so `updateTeams` is only called whenever the user, instead of simply anyone, joins or leaves a tournament.